src/psyclone/psyir/nodes/acc_directives.py

```diff
@@ -275,6 +306,46 @@ class ACCParallelDirective(ACCRegionDirective):
     a DataDirective.
 
     '''
+    def __init__(self, children=None, parent=None, async_queue=False):
+        super().__init__(children=children, parent=parent)
+        self.async_queue = async_queue
+
+    def __eq__(self, other):
+        '''
+        Checks whether two nodes are equal. Two ACCParallelDirective nodes are
+        equal if their default_present members are equal.
+
+        :param object other: the object to check equality to.
+
+        :returns: whether other is equal to self.
+        :rtype: bool
+        '''
+        is_eq = super().__eq__(other)
+        is_eq = is_eq and self.async_queue == other.async_queue
+
+        return is_eq
+
+    @property
+    def async_queue(self):
```

Comment:
I see we've got quite a bit of duplication of this new `async_queue` functionality. I've had a look at the inheritance diagram:
![image](https://github.com/stfc/PSyclone/assets/17409212/491cfbd1-7bd1-4d6e-858e-133656510277)
and can see that, while I think we could add it to `ACCRegionDirective` we'd still have to separately add it to `ACCEnterDataDirective` and `ACCUpdateDirective`. Therefore, I think it would be best if we created a new 'mixin' class that provides this functionality to those classes that need it. Would you be OK to do that? (That class could also have the `_build_async_string` functionality as a method.)